### PR TITLE
issue #72 solved

### LINKED ISSUE #72: migrate javascript files in dashboard to typescript

### PR DESCRIPTION
**Pull request by bosiljkatodic00**
# Goal
resolves #72 
All JavaScript files from dashboard have been migrated to TypeScript.

# Checklist

- [x] This PR has been connected to the issue number #72

### CODE REVIEW

**Review by JerryJeager — CHANGES_REQUESTED** [review]
also resolve merge conflicts here, thanks

File: src/components/dashboard/EventForm.tsx [hunk 1]
```diff
@@ -301,26 +365,33 @@ function StepTwo({ handleSubmit, handleImageUpload }) {
...
             <div className='text-center'>
                 <h1 className='text-4xl font-semibold'>Event Created Successfully!</h1>
                 <p className='text-[#9A9A9A]'>Copy to share the link with your attendees</p>
             </div>
             <div className='flex flex-row'>
-                <TextInput value={eventLink} readOnly={true} />
-                <button onClick={() => {
-                    navigator.clipboard.writeText(eventLink)
-                }} type="submit" className="bg-[#0D35FB] hover:bg-blue-700 text-white text-sm p-4 rounded-md">
+                <TextInput value={eventLink} readOnly={true} width="w-full" />
```
Comment (JerryJeager) [inline src/components/dashboard/EventForm.tsx:382]: vercel deployment failed at the point, can you check it out: 
Screenshot 2024-08-15 043826

**Review by JerryJeager — CHANGES_REQUESTED** [review]

File: src/components/dashboard/EventForm.tsx (hunk 1 above)
Comment (bosiljkatodic00) [reply src/components/dashboard/EventForm.tsx]: I have tried to resolve this, please check it out in last commit.

**Review by JerryJeager — CHANGES_REQUESTED** [review]

File: app/dashboard/events/[id]/guest-list/page.tsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-import { useRouter } from "next/navigation";
-import Searchbar from "../../../../../src/components/Searchbar";
-import AddNewGuest from "../../../../../src/components/dashboard/AddNewGuest";
-import GuestListTable from "../../../../../src/components/dashboard/GuestListTable";
-import { useEffect, useState } from "react";
-import axios from "axios";
+import { useRouter } from 'next/navigation';
+import Searchbar from '../../../../../src/components/Searchbar';
+import {AddNewGuest} from '../../../../../src/components/dashboard/AddNewGuest';
+import { GuestListTable} from '../../../../../src/components/dashboard/GuestListTable';
```
[the remark this answers is not in the document]
Comment (bosiljkatodic00) [reply app/dashboard/events/[id]/guest-list/page.tsx:6]: This one needs to be imported this way.

**Review by JerryJeager — CHANGES_REQUESTED** [review]
Are you able to test the codes locally? There was a typescript issue from the dashboard page....
Screenshot 2024-08-18 062431

**Comment by JerryJeager** [comment]
> Are you able to test the codes locally? There was a typescript issue from the dashboard page.... Screenshot 2024-08-18 062431

the accepted types for guest status are "pending" | "approved" | "rejected" | "accepted"

**Review by JerryJeager — APPROVED** [review]

**Pull request merged**